Tests/scripts/collect_tests/collect_tests.py

```diff
@@ -239,13 +239,21 @@ def _validate_collection(
                     raise ValueError(f'{test} has no path')
                 if PACK_MANAGER.is_test_skipped_in_pack_ignore(playbook_path.name, pack_id):
                     raise SkippedTestException(test, skip_place='.pack_ignore')
-                for integration in test_playbook.implementing_integrations:
+                test_integrations = conf.tests_to_integrations.get(test) or ()  # type: ignore[union-attr]
+                # test_playbook.implementing_integrations is from id set (always empty), test_integrations is from conf.json
+                for integration in test_playbook.implementing_integrations + test_integrations:
                     if reason := conf.skipped_integrations.get(integration):  # type: ignore[union-attr, assignment]
                         raise SkippedTestException(
                             test_name=test,
                             skip_place='conf.json (integrations)',
                             skip_reason=f'{test=} uses {integration=}, which is skipped ({reason=})'
                         )
+                test_marketplaces = conf.tests_to_marketplace_set[test]  # type: ignore[union-attr]
+                logger.info(f'HERE MARKETPLACES: {test_marketplaces=}, {conf.marketplace=}')  # type: ignore[union-attr]
+                if test_marketplaces and conf.marketplace not in test_marketplaces:  # type: ignore[union-attr]
```

Comment:
```suggestion
                if test_marketplaces and (conf.marketplace not in test_marketplaces):  # type: ignore[union-attr]
```
just for readability